### ISSUE TITLE
The examples of the customised python backend are not working with the inference server. Cannot run interpreter host. Errno = 2

### ISSUE THREAD

**Issue by mostafa-mahmoud** [triton-inference-server/server#2671]
### Description
When I run the examples of customized python backend, triton server terminates with the error `Cannot run interpreter host. Errno = 2`. There are no helpful error messages.

### Triton Information

Triton version: 2.3.0
Docker image version: 20.09
Nvidia Driver: 450.66
OS: Ubuntu 16.04

I am using the prebuilt docker image as seen in the command below.

### To Reproduce

First of all, the files' structure (taken from the [example](https://github.com/triton-inference-server/python_backend/tree/main/examples/add_sub)) of the working directory before running the command has the tree:
1. `config.pbtxt` copied from the example [config.pbtxt](https://raw.githubusercontent.com/triton-inference-server/python_backend/main/examples/add_sub/config.pbtxt):
2. `1/model.py` copied from the example [model.py](https://raw.githubusercontent.com/triton-inference-server/python_backend/main/examples/add_sub/model.py):

```
.
└── add_sub
    ├── 1
    │   └── model.py
    └── config.pbtxt

2 directories, 2 files
```

(The issue also occurs by using the pytorch [example](https://github.com/triton-inference-server/python_backend/tree/main/examples/pytorch) in the same structure)

To reproduce the error, I simply run the docker container using the command:
```
nvidia-docker run --rm -e CUDA_VISIBLE_DEVICES=0 --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 -p 8000:8000 -v $PWD:/models nvcr.io/nvidia/tritonserver:20.09-py3 tritonserver --model-store=/models
```
which gives the output:
```
=============================
== Triton Inference Server ==
=============================

NVIDIA Release 20.09 (build 16016295)

Copyright (c) 2018-2020, NVIDIA CORPORATION.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.
NVIDIA modifications are covered by the license terms that apply to the underlying
project or file.

I0327 10:41:01.946276 1 metrics.cc:184] found 3 GPUs supporting NVML metrics
I0327 10:41:01.952288 1 metrics.cc:193]   GPU 0: GeForce GTX 1080 Ti
I0327 10:41:01.958021 1 metrics.cc:193]   GPU 1: GeForce GTX 1080 Ti
I0327 10:41:01.963681 1 metrics.cc:193]   GPU 2: GeForce GTX 1080 Ti
I0327 10:41:01.963788 1 server.cc:120] Initializing Triton Inference Server
I0327 10:41:01.963799 1 server.cc:121]   id: 'triton'
I0327 10:41:01.963805 1 server.cc:122]   version: '2.3.0'
I0327 10:41:01.963812 1 server.cc:128]   extensions:  classification sequence model_repository schedule_policy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data statistics
I0327 10:41:02.154165 1 pinned_memory_manager.cc:195] Pinned memory pool is created at '0x7ff476000000' with size 268435456
I0327 10:41:02.154736 1 cuda_memory_manager.cc:98] CUDA memory pool is created on device 0 with size 67108864
I0327 10:41:02.156760 1 model_repository_manager.cc:714] loading: add_sub:1
E0327 10:41:02.167339 49 python.cc:261] Cannot run interpreter host. Errno = 2
python_interpreter_path: python3
python_interpreter_startup: /opt/tritonserver/lib/python/runtime/startup.py
pymodule_path_: /models/add_sub/1/model.py
instance_name: add_sub_0

E0327 10:41:07.169617 1 model_repository_manager.cc:899] failed to load 'add_sub' version 1: Internal: failed to connect to all addresses
I0327 10:41:07.169832 1 server.cc:213] Waiting for in-flight requests to complete.
I0327 10:41:07.169934 1 server.cc:228] Timeout 30: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models
```

An alternative way to reproduce the error, is running:
```
nvidia-docker run --rm -e CUDA_VISIBLE_DEVICES=0 --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 -p 8000:8000 -v $PWD:/models -it nvcr.io/nvidia/tritonserver:20.09-py3
```
then inside the container running `tritonserver --model-store=/models`, which gives the output:
```
root@75baa2f4ac01:/opt/tritonserver# tritonserver --model-store=/models
I0327 10:59:09.575045 58 metrics.cc:184] found 3 GPUs supporting NVML metrics
I0327 10:59:09.580685 58 metrics.cc:193]   GPU 0: GeForce GTX 1080 Ti
I0327 10:59:09.586582 58 metrics.cc:193]   GPU 1: GeForce GTX 1080 Ti
I0327 10:59:09.592319 58 metrics.cc:193]   GPU 2: GeForce GTX 1080 Ti
I0327 10:59:09.592530 58 server.cc:120] Initializing Triton Inference Server
I0327 10:59:09.592550 58 server.cc:121]   id: 'triton'
I0327 10:59:09.592561 58 server.cc:122]   version: '2.3.0'
I0327 10:59:09.592573 58 server.cc:128]   extensions:  classification sequence model_repository schedule_policy model_configuration system_shared_memory cuda_shared_memory binary_tensor_data statistics
I0327 10:59:09.803616 58 pinned_memory_manager.cc:195] Pinned memory pool is created at '0x7f7d90000000' with size 268435456
I0327 10:59:09.804166 58 cuda_memory_manager.cc:98] CUDA memory pool is created on device 0 with size 67108864
I0327 10:59:09.806189 58 model_repository_manager.cc:714] loading: add_sub:1
E0327 10:59:09.816343 64 python.cc:261] Cannot run interpreter host. Errno = 2
python_interpreter_path: python3
python_interpreter_startup: /opt/tritonserver/lib/python/runtime/startup.py
pymodule_path_: /models/add_sub/1/model.py
instance_name: add_sub_0

Terminated
```

----------------------------
Just for reference (in case the content of the links above gets updated later):
The content of `config.pbtxt`:

```
name: "ASR"
backend: "python"
input {
  name: "the_input"
  data_type: TYPE_FP32
  dims: [-1, 1]
}
output {
  name: "sentiment_score"
  data_type: TYPE_FP32
  dims: [2]
}
instance_group [{ kind: KIND_CPU }]
max_batch_size: 128
dynamic_batching {
  preferred_batch_size: [1, 32, 64, 128]
  max_queue_delay_microseconds: 10
}
```

The content of `1/model.py`:

```python
# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions
# are met:
#  * Redistributions of source code must retain the above copyright
#    notice, this list of conditions and the following disclaimer.
#  * Redistributions in binary form must reproduce the above copyright
#    notice, this list of conditions and the following disclaimer in the
#    documentation and/or other materials provided with the distribution.
#  * Neither the name of NVIDIA CORPORATION nor the names of its
#    contributors may be used to endorse or promote products derived
#    from this software without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

import numpy as np
import sys
import json

# triton_python_backend_utils is available in every Triton Python model. You
# need to use this module to create inference requests and responses. It also
# contains some utility functions for extracting information from model_config
# and converting Triton input/output types to numpy types.
import triton_python_backend_utils as pb_utils


class TritonPythonModel:
    """Your Python model must use the same class name. Every Python model
    that is created must have "TritonPythonModel" as the class name.
    """

    def initialize(self, args):
        """`initialize` is called only once when the model is being loaded.
        Implementing `initialize` function is optional. This function allows
        the model to intialize any state associated with this model.

        Parameters
        ----------
        args : dict
          Both keys and values are strings. The dictionary keys and values are:
          * model_config: A JSON string containing the model configuration
          * model_instance_kind: A string containing model instance kind
          * model_instance_device_id: A string containing model instance device ID
          * model_repository: Model repository path
          * model_version: Model version
          * model_name: Model name
        """

        # You must parse model_config. JSON string is not parsed here
        self.model_config = model_config = json.loads(args['model_config'])

        # Get OUTPUT0 configuration
        output0_config = pb_utils.get_output_config_by_name(
            model_config, "OUTPUT0")

        # Get OUTPUT1 configuration
        output1_config = pb_utils.get_output_config_by_name(
            model_config, "OUTPUT1")

        # Convert Triton types to numpy types
        self.output0_dtype = pb_utils.triton_string_to_numpy(
            output0_config['data_type'])
        self.output1_dtype = pb_utils.triton_string_to_numpy(
            output1_config['data_type'])

    def execute(self, requests):
        """`execute` MUST be implemented in every Python model. `execute`
        function receives a list of pb_utils.InferenceRequest as the only
        argument. This function is called when an inference request is made
        for this model. Depending on the batching configuration (e.g. Dynamic
        Batching) used, `requests` may contain multiple requests. Every
        Python model, must create one pb_utils.InferenceResponse for every
        pb_utils.InferenceRequest in `requests`. If there is an error, you can
        set the error argument when creating a pb_utils.InferenceResponse

        Parameters
        ----------
        requests : list
          A list of pb_utils.InferenceRequest

        Returns
        -------
        list
          A list of pb_utils.InferenceResponse. The length of this list must
          be the same as `requests`
        """

        output0_dtype = self.output0_dtype
        output1_dtype = self.output1_dtype

        responses = []

        # Every Python backend must iterate over everyone of the requests
        # and create a pb_utils.InferenceResponse for each of them.
        for request in requests:
            # Get INPUT0
            in_0 = pb_utils.get_input_tensor_by_name(request, "INPUT0")
            # Get INPUT1
            in_1 = pb_utils.get_input_tensor_by_name(request, "INPUT1")

            out_0, out_1 = (in_0.as_numpy() + in_1.as_numpy(),
                            in_0.as_numpy() - in_1.as_numpy())

            # Create output tensors. You need pb_utils.Tensor
            # objects to create pb_utils.InferenceResponse.
            out_tensor_0 = pb_utils.Tensor("OUTPUT0",
                                           out_0.astype(output0_dtype))
            out_tensor_1 = pb_utils.Tensor("OUTPUT1",
                                           out_1.astype(output1_dtype))

            # Create InferenceResponse. You can set an error here in case
            # there was a problem with handling this inference request.
            # Below is an example of how you can set errors in inference
            # response:
            #
            # pb_utils.InferenceResponse(
            #    output_tensors=..., TritonError("An error occured"))
            inference_response = pb_utils.InferenceResponse(
                output_tensors=[out_tensor_0, out_tensor_1])
            responses.append(inference_response)

        # You should return a list of pb_utils.InferenceResponse. Length
        # of this list must match the length of `requests` list.
        return responses

    def finalize(self):
        """`finalize` is called only once when the model is being unloaded.
        Implementing `finalize` function is OPTIONAL. This function allows
        the model to perform any necessary clean ups before exit.
        """
        print('Cleaning up...')
```

**Comment by Tabrizian** [comment]
@mostafa-mahmoud Python backend was released in 20.10 release. Can you try 21.03 release?

**Comment by mostafa-mahmoud** [comment]
@Tabrizian It worked with that version, thank you! I just need to update the driver for it.
